src/hotspot/share/prims/jvmtiEnvBase.hpp

```diff
@@ -171,6 +171,17 @@ class JvmtiEnvBase : public CHeapObj<mtInternal> {
   // Otherwise return JNIHandles::resolve_external_guard(thread).
   static oop current_thread_obj_or_resolve_external_guard(jthread thread);
 
+  static bool is_JavaThread_current(JavaThread* jt, oop thr_obj) {
+    if (jt == NULL) {
+      return false; // jt can be NULL in case of a virtual thread
+    }
+    JavaThread* current = JavaThread::current();
+    oop cur_obj = current->jvmti_vthread();
+    bool is_current = jt == current && (cur_obj == NULL || cur_obj == thr_obj);
```

Comment:
It might be better to restructure this "if" and check for jt==current before we ask for cur_obj, or at least add brackets.